### PR TITLE
fix load balancer IP change when updating service

### LINKED ISSUE #66424: Azure - ILB Frontend IP configuration does not change while updating service
/kind bug

**What happened**:
ILB Frontend IP configuration does not change while updating service
**What you expected to happen**:
LB Frontend IP configuration should get updated
**How to reproduce it (as minimally and precisely as possible)**:
Source issue : https://github.com/Azure/AKS/issues/532
AKS service is created using .yaml manifest. ILB is described as it is shown in the documentation:

annotations:
    service.beta.kubernetes.io/azure-load-balancer-internal: "true"
spec:
  ports:
  - port: 80
  type: LoadBalancer
  loadBalancerIP: 10.240.200.2
ILB instance with the "kubernetes-internal" name is created with a correct private IP address pointing to the service described. If I change the "loadBalancerIP" to another internal IP address and re-deploy the service, the last is deployed with the correct new IP address, but "kubernetes-internal" ILB still has the old Frontend IP configuration.

**Environment**:
- Cloud provider or hardware configuration:
Azure

### PR DESCRIPTION
**Pull request by fseldow**
**What this PR does / why we need it**:
To fix bug in azure load balancer. Change frontend IP Configuration while updating service without changing its internal annotation

**Which issue(s) this PR fixes** :
Fixes #66424 

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_loadbalancer.go [hunk 1]
```diff
@@ -496,6 +506,24 @@ func getIdleTimeout(s *v1.Service) (*int32, error) {
...
 }
 
+func judgeDeleteFrontendIPConfiguration(config network.FrontendIPConfiguration, service *v1.Service, lbFrontendIPConfigName string) bool {
+	if serviceOwnsFrontendIP(config, service) && !strings.EqualFold(to.String(config.Name), lbFrontendIPConfigName) {
+		return true
+	}
+	if !strings.EqualFold(to.String(config.Name), lbFrontendIPConfigName) {
+		return false
+	}
+	loadBalancerIP := service.Spec.LoadBalancerIP
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_loadbalancer.go]: It should also be checked with config.PublicIPAddress. If loadBalancerIP == config.PublicIPAddress, then the frontendIPConfiguration should be reserved.

File: pkg/cloudprovider/providers/azure/azure_loadbalancer.go [hunk 2]
```diff
@@ -496,6 +506,24 @@ func getIdleTimeout(s *v1.Service) (*int32, error) {
 	return &to32, nil
 }
 
+func judgeDeleteFrontendIPConfiguration(config network.FrontendIPConfiguration, service *v1.Service, lbFrontendIPConfigName string) bool {
```
Comment (feiskyer) [inline pkg/cloudprovider/providers/azure/azure_loadbalancer.go]: Change to `isFrontendIPChanged()` for clear?

**Comment by fseldow** [comment]
@feiskyer Hi, could you please have a review on my PR? Sorry that I nearly forgot this PR.

**Comment by feiskyer** [comment]
Looks good to me in general. Please address nits above.

**Pull request merged**